include/seqan3/alignment/band/static.hpp

```diff
@@ -0,0 +1,146 @@
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides seqan3::band_static.
+ * \author Jörg Winkler <j.winkler AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/band/detail.hpp>
+#include <seqan3/core/detail/strong_type.hpp>
+
+namespace seqan3
+{
+
+/*!\brief Type for the lower diagonal in a banded alignment.
+ *\ingroup alignment
+ */
+template <typename value_t>
+requires std::is_integral_v<value_t>
+struct lower_bound : detail::strong_type<value_t, lower_bound<value_t>>
+{
+    //!\brief Inheriting constructors from base class.
+    using detail::strong_type<value_t, lower_bound<value_t>>::strong_type;
+};
+
+/*!\brief Type for the upper diagonal in a banded alignment.
+ *\ingroup alignment
+ */
+template <typename value_t>
+requires std::is_integral_v<value_t>
+struct upper_bound : detail::strong_type<value_t, upper_bound<value_t>>
+{
+    //!\brief Inheriting constructors from base class.
+    using detail::strong_type<value_t, upper_bound<value_t>>::strong_type;
+};
+
+/*!\name Deduction guides
+ * \brief Deduces template parameter from the argument.
+ * \relates seqan3::lower_bound
+ * \{
+ */
+template <typename value_t>
+requires std::is_integral_v<value_t>
+lower_bound(value_t) -> lower_bound<value_t>;
+//!\}
+
+
+/*!\name Deduction guides
+ * \brief Deduces template parameter from the argument.
+ * \relates seqan3::upper_bound
+ * \{
+ */
+
+template <typename value_t>
+requires std::is_integral_v<value_t>
+upper_bound(value_t) -> upper_bound<value_t>;
+//!\}
+
+
+/*!\brief Data structure for a static band.
+ * \ingroup alignment
+ * \tparam value_t The value type for the diagonal indices.
```

Comment:
add `; std::is_integral_v<valut_t> must evaluate to true.` or similar sentence.
or if you see the comment below: Must model std::Integral.